addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXBridgeBaseThingHandler.java

```diff
@@ -0,0 +1,974 @@
+
+            if (getThing().getStatus() == ThingStatus.ONLINE && pc != null) {
+
+                RetryDatapoint datapoint = readDatapoints.poll();
+
+                if (datapoint != null) {
+                    datapoint.incrementRetries();
+
+                    boolean success = false;
+                    try {
+                        logger.trace("Sending read request on the KNX bus for datapoint {}",
+                                datapoint.getDatapoint().getMainAddress());
+                        pc.read(datapoint.getDatapoint());
+                        success = true;
+                    } catch (KNXException e) {
+                        logger.warn("Cannot read value for datapoint '{}' from KNX bus: {}",
+                                datapoint.getDatapoint().getMainAddress(), e.getMessage());
+                    } catch (KNXIllegalArgumentException e) {
+                        logger.warn("Error sending KNX read request for datapoint '{}': {}",
+                                datapoint.getDatapoint().getMainAddress(), e.getMessage());
+                    } catch (InterruptedException e) {
+                        logger.warn("Error sending KNX read request for datapoint '{}': {}",
+                                datapoint.getDatapoint().getMainAddress(), e.getMessage());
+                    }
+                    if (!success) {
+                        if (datapoint.getRetries() < datapoint.getLimit()) {
+                            logger.debug(
+                                    "Adding the read request (after attempt '{}') for datapoint '{}' at position '{}' in the queue",
+                                    datapoint.getRetries(), datapoint.getDatapoint().getMainAddress(),
+                                    readDatapoints.size() + 1);
+                            readDatapoints.add(datapoint);
+                        } else {
+                            logger.debug("Giving up reading datapoint {} - nubmer of maximum retries ({}) reached.",
+                                    datapoint.getDatapoint().getMainAddress(), datapoint.getLimit());
+                        }
+                    }
+                }
+            }
+        }
+    };
+
+    public class RetryDatapoint {
+
+        private Datapoint datapoint;
+        private int retries;
+        private int limit;
+
+        public Datapoint getDatapoint() {
+            return datapoint;
+        }
+
+        public int getRetries() {
+            return retries;
+        }
+
+        public void incrementRetries() {
+            this.retries++;
+        }
+
+        public int getLimit() {
+            return limit;
+        }
+
+        public RetryDatapoint(Datapoint datapoint, int limit) {
+            this.datapoint = datapoint;
+            this.retries = 0;
+            this.limit = limit;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            return (o instanceof RetryDatapoint)
```

Comment:
again, not null-safe and `hashCode()` is missing!

PS: let eclipse generate it... ;-)